### ISSUE TITLE
MissingParameter error. 

### ISSUE THREAD

**Issue by nguyenthuan1** [jordanclark/cfamazonproductad#1]
When I tried this on my coldfusion server, I keep getting this error. Anyone else having this error?

{"__type":"com.amazon.paapi5#ValidationException","Errors":[{"Code":"MissingParameter","Message":"PartnerTag should be provided."},{"Code":"MissingParameter","Message":"PartnerType should be provided."}]}
on the index.cfm page this is how I called it. 
<cfobject name="myObj" component="amazonProductAd5">
<cfset test= myObj.init().usDefaults().SearchItems( "harry", "ItemInfo.Title" )>
<cfdump var =#test#>
I hard coded the secret key , partnertag, and access key on the .cfc page.

**Comment by jordanclark** [comment]
What server are you using?

What is myObj? Normally I think you would create the object like this:
```
<cfset myObj = new cfamazonproductad.amazonproductad5(
	accessKeyId= "..."
,	secretAccessKey= "..."
,	partnerTag= "..."
).usDefaults()>
```

Or what I do is make my own personalized object that extends amazonproductad5 and it embeds the accessKeyId / secretAccessKey / partnerTag in the init method like this:

```
component extends= "cfamazonproductad.amazonProductAd5" {
	function init() {
		super.init(
			accessKeyId= "..."
		,	secretAccessKey= "..."
		,	partnerTag= "..."
		).usDefaults();
		return this;
	}
}
```

Then just do:
```
test = new myAmazon5().searchItems( "harry", "ItemInfo.Title" );
```

**Comment by nguyenthuan1** [comment]
I have coldfusion 18 development. myObj is declared by using cfobject in the index.cfm file.
 It complains about a couple things. 

1.		In the parameter,  it complains about it not being a Boolean. I replaced it with a number 0 and the error went away. 
		boolean debug= ( request.debug ?: false )
2.		 There is an extra comma at the end of the parameter? I removed that comma and then the error went away. line 19
3.	Complex object types cannot be converted to simple values.
	451 : 		if( len( arguments.LanguagesOfPreference ) && isSimpleValue( arguments.LanguagesOfPreference ) ) {
	452 : 		 	arguments.LanguagesOfPreference= listToArray( arguments.LanguagesOfPreference, this.defaultDelim );
	453 : 		}
I removed the [] on line 251 in this.defaultLanguage= arguments.language ; and then the error went away. The parameter passing is a string that's why it complained.
4. Once all those errors go away, it gives me to parameter missing error as stated. keeps saying partnerTag and type is missing. 
Note: coldfusion complained about cfprocessingdirective( preserveCase=true ); . I had to comment it out for the error to go away.
